Source/GameView/Repositories/GameDetailsRepository/GameDetailsAPIRepo.swift

```diff
@@ -0,0 +1,175 @@
+        serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+          
+          guard let self = self else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          guard let game = dataBundle.extraData[gameDataBundleKeys.game] as? Game else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          self.decodeGameDetails(using: data, basedOn: game)
+          
+          dispatchGroup.leave()
+          
+        }
+        
+        // MARK: - ---------------------------
+
+        // MARK: - Call Failed Closure
+        serviceCaller.callFailed = { error in
+          dispatchGroup.leave()
+        }
+        
+        // MARK: - ---------------------------
+        
+        let urlString = "\(SteamURLComponents.specificGameURL)\(game.appID)"
+        let url: URL = URL(string: urlString)!
+        
+        do {
+          try serviceCaller.makeServiceCall(with: url, and: dataBundle)
+        } catch {
+          dispatchGroup.leave()
+          fatalError("Service call not set up properly")
+        }
+        
+      })
+      
+      dispatchGroup.wait()
+      
+    }
+    
+    return serviceCallOperation
+    
+  }
+  
+  private func notifyGameDetailsLoaded(withData gameList: [Game]) {
+    observer?.gameDetailsFinishedLoading(withData: gameList)
+  }
+  
+  private func extractGameData(detailsWrapper: [String: Any], game: Game) {
+
+    guard let gameDetailsData = detailsWrapper[gameDetailsJSONResponseKeys.data] as? [String: Any] else {
+      return
+    }
+
+    if let priceOverview = gameDetailsData[gameDetailsJSONResponseKeys.priceOverview] as? [String: Any] {
+      game.price = Double((priceOverview[gameDetailsJSONResponseKeys.finalPriceOverview] as? Int ?? 0)/100)
+    }
+    
+    if let shortDescription = gameDetailsData[gameDetailsJSONResponseKeys.shortDescription] as? String {
+      game.shortDescription = shortDescription
+    }
+    
+    if let developers = gameDetailsData[gameDetailsJSONResponseKeys.developers] as? Array<String> {
+      game.developers = developers
+    }
+    
+    if let publishers = gameDetailsData[gameDetailsJSONResponseKeys.publishers] as? Array<String> {
+      game.publishers = publishers
+    }
+    
+    if let headerImagePath = gameDetailsData[gameDetailsJSONResponseKeys.headerImage] as? String {
+      game.headerImagePath = headerImagePath
+    }
+    
+  }
+  
+  private func decodeGameDetails(using data: Data, basedOn game: Game) {
+    
+    do {
+
+      //Wanted data is wrapped in extra keys. This unwraps its.
+      let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
+      let jsonWrapper = json as? [String: Any]
+      let mainResponseBody = jsonWrapper?[game.appID] as? [String: Any]
+
+      if let mainResponseBody = mainResponseBody {
+        self.extractGameData(detailsWrapper: mainResponseBody, game: game)
+      }
+
+    } catch let err {
+      print(err)
```

Comment:
Is there any way we could handle this error instead of just printing ?

Reply:
I can remove the print and do nothing for now, error propagation is planned. Due to my lack of understanding I didn't create a full way for errors to propagate to the front. I'll have to do this through the entire application.

Reply:
It's up to you really, approving. Nothing wrong with just printing if you plan to look at handling it better later